review-service/controller/review-controller.js

```diff
@@ -0,0 +1,138 @@
+import {
+  writeScoresToCache,
+  getScoresFromCache,
+  deleteScoresFromCache,
+} from '../model/redis-repository.js'
+
+import {
+  ormGetAllScoresForUser as _getUserScores,
+  ormCreateScores as _createScores,
+  ormDeleteAllScoresForUser as _deleteUserScores,
+} from '../model/review-orm.js'
+
+// for extensibility, if we want to have more criteria, just add here!
+export const FIELDS = [
+  'codeCorrectness',
+  'codeDesign',
+  'codeStyle',
+  'communicationStyle',
+  'timeManagement',
+]
+
+const generateZeroArray = (len) => {
+  return Array.from(Array(len), () => 0)
+}
+
+const generateAverageScoresForUser = (scores) => {
+  const res = {
+    userId: scores.userId,
+    totalReviews: scores.totalReviews,
+  }
+  const roundUpToTwoDecimals = (n) => {
+    return Math.round((n / scores.totalReviews) * 100) / 100
+  }
+  FIELDS.map((field) => {
+    res[field] = roundUpToTwoDecimals(scores[field])
+  })
+  return res
+}
+
+const generateTotalScoresForUser = (reviews) => {
```

Comment:
I think this function is supposed to take in a second argument `userId` (line 70) and use it on line 48 cos otherwise `reviews.userId` seems to be undefined when I was testing